rust/lance-index/src/vector/ivf/shuffler.rs

```diff
@@ -148,25 +101,94 @@ pub async fn shuffle_dataset(
         ),
     ]));
 
-    let stream = lance_core::io::RecordBatchStreamAdapter::new(schema.clone(), stream);
+    // step1: either use precomputed shuffle files or write shuffle data to a file
+    let shuffler = if let Some((path, buffers)) = precomputed_shuffle_buffers {
+        let mut shuffler = IvfShuffler::try_new(
+            num_partitions,
+            num_sub_vectors,
+            Some(path),
+            Schema::try_from(schema.as_ref())?,
+        )?;
+        unsafe {
+            shuffler.set_unsorted_buffers(&buffers);
+        }
 
-    let mut shuffler = IvfShuffler::try_new(
-        num_partitions,
-        num_sub_vectors,
-        None,
-        Schema::try_from(schema.as_ref())?,
-    )?;
+        shuffler
+    } else {
+        let mut shuffler = IvfShuffler::try_new(
```

Comment:
This is the old `(row_id, part_id)` behavior?

Let's make a TODO / issue to remove it?